assignments/prototypes.js

```diff
@@ -1,28 +1,49 @@
 /*
   Object oriented design is commonly used in video games.  For this part of the assignment you will be implementing several constructor functions with their correct inheritance hierarchy.
 
-  In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.  
+  In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.
 
   At the bottom of this file are 3 objects that all end up inheriting from Humanoid.  Use the objects at the bottom of the page to test your constructor functions.
-  
+
   Each constructor function has unique properties and methods that are defined in their block comments below:
 */
-  
+
 /*
   === GameObject ===
   * createdAt
   * name
   * dimensions (These represent the character's size in the video game)
   * destroy() // prototype method that returns: `${this.name} was removed from the game.`
 */
+  function GameObject(attrs) {
+    this.createdAt = attrs.createdAt,
+ this.dimensions = attrs.dimensions,
+this.destroy = function() {
+  return `${this.name} was removed from the game.`;
+}
+}
 
```

Comment:
So this works, and I actually did the same thing on my solution back in the day. The issue with this is everytime we call the constructor function, it will create a new copy of the `destroy` function, which is going to waste memory. The advantage of prototypes is that we can declare that function once and then have every instance of `GameObject` use that same function.

We do that by adding the function as a property on the object's `prototype`.
`GameObject.prototype.destroy = function() { return `${this.....}`}`